src/lib/docker-containers.mjs

```diff
@@ -0,0 +1,325 @@
+import { container } from "../cli.mjs";
+import { CommandError } from "./errors.mjs";
+
+const IMAGE_NAME = "fauna/faunadb:latest";
+
+/**
+ * Ensures the container is running
+ * @param {string} imageName The name of the image to create the container from
+ * @param {string} containerName The name of the container to start
+ * @param {number} hostPort The port on the host machine mapped to the container's port
+ * @param {number} containerPort The port inside the container Fauna listens on
+ * @param {boolean} pull Whether to pull the latest image
+ * @returns {Promise<void>}
+ */
+export async function ensureContainerRunning({
+  containerName,
+  hostPort,
+  containerPort,
+  pull,
+}) {
+  const logger = container.resolve("logger");
+  if (pull) {
+    await pullImage(IMAGE_NAME);
+  }
+  const logStream = await startContainer({
+    imageName: IMAGE_NAME,
+    containerName,
+    hostPort,
+    containerPort,
+  });
+  logger.stderr(
+    `[StartContainer] Container '${containerName}' started. Monitoring HealthCheck for readiness.`,
+  );
+  await waitForHealthCheck({
+    url: `http://localhost:${hostPort}`,
+    logStream,
+  });
+  logger.stderr(
+    `[ContainerReady] Container '${containerName}' is up and healthy.`,
+  );
```

Comment:
not for this PR, but this is probably a good fit for one of the verbosity level loggers (`logger.debug/info/warn/error/fatal`). maybe we put it on info and change the default log level to include info? going through the verbosity loggers gives users easier / better sifting (b/c you give it a component and a log level, and they can use those to filter in/out sets of logs).